File: app/src/main/java/com/example/dagger_hilt/ui/activity/singup/SingInViewModel.java
package com.example.dagger_hilt.ui.activity.singup;

import androidx.lifecycle.ViewModel;

import com.example.dagger_hilt.domain.StorageRepository;

import javax.inject.Inject;

import dagger.hilt.android.lifecycle.HiltViewModel;

@HiltViewModel
public class SingInViewModel extends ViewModel {
    private final StorageRepository storageRepository;

    @Inject
    public SingInViewModel(StorageRepository storageRepository) {
        this.storageRepository = storageRepository;
    }

    public void singIn(String userName, String password) {
        storageRepository.registerUser(userName, password);
    }
}

File: app/src/main/java/com/example/dagger_hilt/ui/activity/singup/SingInViewModel.kt
package com.example.dagger_hilt.ui.activity.singup

import androidx.lifecycle.ViewModel
import com.example.dagger_hilt.domain.StorageRepository
import dagger.hilt.android.lifecycle.HiltViewModel
import javax.inject.Inject

@HiltViewModel
class SingInViewModel @Inject constructor(private val storageRepository: StorageRepository) :
    ViewModel() {
    fun singIn(userName: String, password: String) = storageRepository.registerUser(userName, password)
}